### PR TITLE
Pure Terraform CICD improvements

### PR DESCRIPTION
**Pull request by umairidris**
- Port validate to pure TF
- Make plan work with deletion check
- Move APIs to devops
- grant CICD roles/viewer on devops

### CODE REVIEW

File: docs/tfengine/recipes/cicd.md [hunk 1]
```diff
@@ -65,16 +65,15 @@ Type: string
...
 NOTE: The modules will be deployed in the given order. If a module
 depends on another module, it should show up after it in this list.
 
-Type: array(string)
-
-### managed_services
-
-APIs to enable in the devops project so the Cloud Build service account
-can manage those services in other projects.
+NOTE: The CICD has permission to update APIs within its own project.
```
Comment (xingao267) [inline docs/tfengine/recipes/cicd.md:73]: The CICD only has permission to update ...

File: examples/tfengine/generated/org_foundation/cicd/main.tf [hunk 2]
```diff
@@ -69,6 +69,7 @@ locals {
   cloudbuild_devops_roles = [
     # Enable Cloud Build SA to list and enable APIs in the devops project.
     "roles/serviceusage.serviceUsageAdmin",
+    "roles/viewer",
```
Comment (xingao267) [inline examples/tfengine/generated/org_foundation/cicd/main.tf:72]: Add a comment why viewer role is needed

File: examples/tfengine/generated/team/bootstrap/main.tf [hunk 3]
```diff
@@ -43,6 +43,12 @@ module "project" {
   skip_gcloud_download    = true
   activate_apis = [
     "cloudbuild.googleapis.com",
+    "container.googleapis.com",
```
Comment (xingao267) [inline examples/tfengine/generated/team/bootstrap/main.tf:46]: remove these APIs from the CICD module?

Comment (xingao267) [reply examples/tfengine/generated/team/bootstrap/main.tf:46]: nvm, you are just adding the additional used service APIs here.

Comment (umairidris) [reply examples/tfengine/generated/team/bootstrap/main.tf:46]: Right, that way it's consistent experience when using CICD and not using CICD to enable APIs on the devops project.

File: templates/tfengine/components/cicd/configs/tf-apply.yaml [hunk 4]
```diff
@@ -12,6 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Note: Terraform version used in the automation is 0.12.29.
+{{- $cft := "gcr.io/cloud-foundation-cicd/cft/developer-tools@sha256:47db3e958fbaa0d95881cf99501ebf4522a261ddc68d9566b7cf70e26cf7cddb"}}
```
Comment (MartinPetkov) [inline templates/tfengine/components/cicd/configs/tf-apply.yaml:16]: Change the rest of the file to use this variable.

Comment (umairidris) [reply templates/tfengine/components/cicd/configs/tf-apply.yaml:16]: good catch

**Pull request merged**